lib/debugmenu.js

```diff
@@ -41,182 +49,50 @@ class DebugMenu extends LitElement {
     return {
-      debugEntries: { type: Object },
       isOpen: { type: Boolean },
-      cursorBlock: { type: Object }
+      cursorBlock: { type: Object },
+      bot: { type: Object }
     }
   }
 
   render () {
-    return html`
-    <div id="debugmenu-wrapper" class="debugmenu-wrapper">
-      <div class="debugmenu" id="debugmenu" style="display: none;">
-        <p id="debug-entry-renderer">Renderer: three.js r${global.THREE.REVISION}</p>
-        </br>
-        ${Object.entries(this.debugEntries).map(([id, entry]) => html`
-          <p id="debug-entry-${id}" style="display:${entry.hidden ? 'none' : 'block'}">${entry.text}</p>${entry.breaking ? html`<br>` : ''}
-        `)}
-      </div>
-    </div>
-    `
-  }
-
-  constructor () {
-    super()
-
-    this.debugEntries = {}
-    this.isOpen = false
-  }
-
-  updated (changedProperties) {
-    if (changedProperties.has('isOpen')) {
-      if (this.isOpen) {
-        this.updateListener = () => {
-          const pos = this.bot.entity.position
-          const rot = [this.bot.entity.yaw, this.bot.entity.pitch]
-
-          const viewDegToMinecraft = (yaw) => {
-            return yaw % 360 - 180 * (yaw < 0 ? -1 : 1)
-          }
-
-          const minecraftYaw = viewDegToMinecraft(rot[0] * -180 / Math.PI)
-          const minecraftQuad = Math.floor(((minecraftYaw + 180) / 90 + 0.5) % 4)
-
-          const target = this.cursorBlock
-          const targetDiggable = (target && this.bot.canDigBlock(target))
-
-          this.updateEntry(this.positionEntry, `XYZ: ${pos.x.toFixed(3)} / ${pos.y.toFixed(3)} / ${pos.z.toFixed(3)}`)
-          this.updateEntry(this.chunkEntry, `Chunk: ${Math.floor(pos.x % 16)} ~ ${Math.floor(pos.z % 16)} in ${Math.floor(pos.x / 16)} ~ ${Math.floor(pos.z / 16)}`)
-
-          const quadsDescription = [
-            'north (towards negative Z)',
-            'east (towards positive X)',
-            'south (towards positive Z)',
-            'west (towards negative X)'
-          ]
-
-          this.updateEntry(this.rotationEntryV, `Facing (viewer): ${rot[0].toFixed(3)} ${rot[1].toFixed(3)}`)
-          this.updateEntry(this.rotationEntryM, `Facing (minecraft): ${quadsDescription[minecraftQuad]} (${minecraftYaw.toFixed(1)} ${(rot[1] * -180 / Math.PI).toFixed(1)})`)
-
-          if (targetDiggable) {
-            if (this.getEntry(this.targetBlockEntry).hidden) this.setEntryVisible(this.targetBlockEntry, true)
-
-            this.updateEntry(this.targetBlockEntry, `Looking at: ${target.position.x} ${target.position.y} ${target.position.z}`)
-          } else if (!targetDiggable && !this.getEntry(this.targetBlockEntry).hidden) {
-            this.setEntryVisible(this.targetBlockEntry, false)
-          }
-        }
-        this.bot.on('move', this.updateListener)
-      } else if (this.updateListener) {
-        this.bot.removeListener('move', this.updateListener)
-        this.updateListener = null
-      }
+    if (!this.isOpen) {
+      return html``
```

Comment:
We can save at least 1+ function call if this gets prevented in shouldUpdate () instead

Reply:
theoritically this makes sense, but I tried it and overriding shouldUpdate to return this.isOpen results in the f3 key not being caught, I'm not 100% sure as to why, so let's keep it like this

Reply:
Probably because you set event listener in `firstUpdated ()` which wouldn't run unless menu is open at that moment